components/brave_shields/resources/panel/components/main-panel/hooks.ts

```diff
@@ -0,0 +1,29 @@
+import * as React from 'react'
+import getPanelBrowserAPI, { UIHandlerReceiver } from '../../api/panel_browser_api'
+
+export function useGetShieldsData() {
+  const [url, setUrl] = React.useState("")
+  const [totalBlockedResources, setTotalBlockedResources] = React.useState(0)
+
+  React.useEffect(() => {
+    const uiHandlerReceiver = new UIHandlerReceiver({
+      onSiteDataChanged: (currentSite) => {
+        setUrl(currentSite.location.url)
+        setTotalBlockedResources(currentSite.totalBlockedResources)
+      }
+    })
+
+    getPanelBrowserAPI().dataHandler.registerUIHandler(
+      uiHandlerReceiver.$.bindNewPipeAndPassRemote())
+
+    // initial
+    getPanelBrowserAPI().dataHandler
```

Comment:
nit: Or maybe you could call `UpdateCurrentSiteData()` from browser process after `UIHandlerReceiver` is bound to avoid duplication (calling same setUrl and setTotal... here and above). It's up to you.

Reply:
nice, this works